clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -10969,6 +10969,9 @@ def err_ivdep_declrefexpr_arg : Error<
 def warn_ivdep_redundant : Warning <"ignoring redundant Intel FPGA loop "
   "attribute 'ivdep': safelen %select{INF|%1}0 >= safelen %select{INF|%3}2">,
   InGroup<IgnoredAttributes>;
+def warn_attribute_spelling_deprecated : Warning<
+  "attribute %0 is deprecated, did you mean to use 'reqd_sub_group_size' instead?">,
```

Comment:
I had suggested making this a warning + note (using something like note_function_suggestion) as it seems to fit how diagnostics work better, and still have a preference for it.  But @elizabethandrews @Fznamznon and @premanandrao can make the final call here.

Reply:
I personally prefer adding a note. However I see instances of 'did you mean' in error/warning message itself in DiagnosticsSemaKinds.td. @erichkeane what does clang community recommend? If they recommend using a separate note, @smanna12 would you mind adding a note? There are examples in DiagnosticsSemaKinds.td

Reply:
I suspect we are consistently inconsistent :)   Clang typically doesn't have its warnings be prescriptive, but I can see that we do in a bunch of cases.  I'm not sure we have official guidance.

That said, if we leave it alone, i think the suggested replacement needs to be a replacement as well.  Finally, don't we mean:

'did you mean to use 'intel::reqd_sub_group_size' instead?"

Reply:
Thanks @elizabethandrews and @erichkeane for the comments. 

>> However I see instances of 'did you mean' in error/warning message itself in DiagnosticsSemaKinds.td.
Yes, i saw several examples that used and so added in the warning message. 

I have added warning+ note and updated all tests with it.